syntax="proto3";

import "Input/proto/input_event.proto";

package astre.ecs;

message InputComponent
{
    // Keys
    repeated astre.input.InputCode pressed = 1;         // All currently pressed keys/buttons
    repeated astre.input.InputCode just_pressed = 2;    // Keys newly pressed this frame
    repeated astre.input.InputCode just_released = 3;   // Keys released this frame

    // Mouse
    astre.input.MouseData mouse = 4;
}